pkg/scheduler/internal/cache/comparer/comparer.go

```diff
@@ -29,31 +29,49 @@ import (
 	internalqueue "k8s.io/kubernetes/pkg/scheduler/internal/queue"
 )
 
-type cacheComparer struct {
-	nodeLister corelisters.NodeLister
-	podLister  corelisters.PodLister
-	cache      schedulerinternalcache.Cache
-	podQueue   internalqueue.SchedulingQueue
-	compareStrategy
+// CacheComparer is an implementation of the Scheduler's cache comparer.
+type CacheComparer struct {
+	CompareStrategy
+
+	NodeLister corelisters.NodeLister
+	PodLister  corelisters.PodLister
+	Cache      schedulerinternalcache.Cache
+	PodQueue   internalqueue.SchedulingQueue
+}
+
+// NewCacheComparer creates a CacheComparer.
+func NewCacheComparer(
```

Comment:
I would prefer this to be named `New`. Since you brought up the idea of sending follow-up PRs to clean this file, I'm okay with this being changed in a different PR.

Reply:
OK